### PR TITLE
Make it possible to retrieve idMap from AST.

### PR DESCRIPTION
**Pull request by Josef-Vonasek**
### Pull Request Description

Make it possible to retrieve idMap from AST.
Added method `children: Seq[T]`
Makes AST comparison ignore AST.ID

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/luna/enso/blob/master/doc/scala-style-guide.md), [Java](https://github.com/luna/enso/blob/master/doc/java-style-guide.md) or [Haskell](https://github.com/luna/enso/blob/master/doc/haskell-style-guide.md) style guides as appropriate.
- [x] All code has been tested where possible.

### CODE REVIEW

File: Syntax/definition/src/main/scala/org/enso/syntax/text/AST.scala [hunk 1]
```diff
@@ -1087,6 +1128,7 @@ object AST {
       segs: Shifted.List1[Match.SegmentOf[T]],
       resolved: AST
     ) extends MacroOf[T] {
+      def children = segs.toList().map(???)
```
Comment (Josef-Vonasek) [inline Syntax/definition/src/main/scala/org/enso/syntax/text/AST.scala]: Does it make sense for `Pattern.MatchOf{T]` to have method `children: Seq[T]`?

File: Syntax/definition/src/main/scala/org/enso/syntax/text/AST.scala [hunk 2]
```diff
@@ -313,27 +319,38 @@ object AST {
...
-  /** Phantom type. Use with care, as Scala cannot prove its proper usage. When
-    * a type is phantom, then its last type argument is not used and we can
-    * safely coerce it to something else.
+  /** Leaf of AST. Should be used only for AST types with no children. Use with
+    * care, since scala cannot infer its right usage.
     */
-  trait Phantom
-  implicit class PhantomOps[T[_] <: Phantom](ident: T[_]) {
-    def coerce[S]: T[S] = ident.asInstanceOf[T[S]]
+  trait Leaf[T] extends ShapeOf[T] {
```
Comment (Josef-Vonasek) [inline Syntax/definition/src/main/scala/org/enso/syntax/text/AST.scala]: Is it possible to remove the parameter T from `Leaf` without breaking inference?

Comment (wdanilo) [reply Syntax/definition/src/main/scala/org/enso/syntax/text/AST.scala]: No

File: Syntax/definition/src/main/scala/org/enso/syntax/text/AST.scala (hunk 1 above)
Comment (wdanilo) [reply Syntax/definition/src/main/scala/org/enso/syntax/text/AST.scala]: See the comments below.

**Pull request closed without merging**